### PR TITLE
Allow font files in sub-folders

### PR DESCRIPTION
**Pull request by felquis**
Hello Guys

I organize my fonts as well:
image

And the actual configuration breaks the build.

```
Warning: Unable to read "dist/styles/fonts/bebasneue" file (Error code: EISDIR).
 Use --force to continue.
Error: EISDIR, illegal operation on a directory
    at Object.fs.readSync (fs.js:476:19)
    at Object.fs.readSync (c:\wamp\www\omo-aba\node_modules\grunt\node_modules\r
imraf\node_modules\graceful-fs\graceful-fs.js:303:23)
    at Object.fs.readFileSync (fs.js:315:28)
    at Object.file.read (c:\wamp\www\omo-aba\node_modules\grunt\lib\grunt\file.j
s:223:19)
    at md5 (c:\wamp\www\omo-aba\node_modules\grunt-rev\tasks\rev.js:20:28)
    at c:\wamp\www\omo-aba\node_modules\grunt-rev\tasks\rev.js:34:20
    at Array.forEach (native)
    at c:\wamp\www\omo-aba\node_modules\grunt-rev\tasks\rev.js:32:20
    at Array.forEach (native)
    at Object.<anonymous> (c:\wamp\www\omo-aba\node_modules\grunt-rev\tasks\rev.
js:31:16)

Aborted due to warnings.
```

---

> That's All folks :pig:

### CODE REVIEW

**Comment by stephenplusplus** [comment]
+1 I think this is a great idea.

**Comment by felipefialho** [comment]
+1 Nice!

File: app/templates/Gruntfile.js [hunk 1]
```diff
@@ -229,7 +229,7 @@ module.exports = function (grunt) {
                         '<%%= yeoman.dist %>/scripts/{,*/}*.js',
                         '<%%= yeoman.dist %>/styles/{,*/}*.css',
                         '<%%= yeoman.dist %>/images/{,*/}*.{png,jpg,jpeg,gif,webp}',
-                        '<%%= yeoman.dist %>/styles/fonts/*'
+                        '<%%= yeoman.dist %>/styles/fonts/**/*.{eot,svg,ttf,woff}'
```
Comment (sindresorhus) [inline app/templates/Gruntfile.js]: use `{,*/}*`

File: app/templates/Gruntfile.js [hunk 2]
```diff
@@ -316,7 +316,7 @@ module.exports = function (grunt) {
                         '*.{ico,png,txt}',
                         '.htaccess',
                         'images/{,*/}*.{webp,gif}',
-                        'styles/fonts/*'
+                        'styles/fonts/**/*.{eot,svg,ttf,woff}'
```
Comment (sindresorhus) [inline app/templates/Gruntfile.js]: use `{,*/}*`

File: app/templates/Gruntfile.js [hunk 3]
```diff
@@ -229,7 +229,7 @@ module.exports = function (grunt) {
                         '<%%= yeoman.dist %>/scripts/{,*/}*.js',
                         '<%%= yeoman.dist %>/styles/{,*/}*.css',
                         '<%%= yeoman.dist %>/images/{,*/}*.{png,jpg,jpeg,gif,webp}',
-                        '<%%= yeoman.dist %>/styles/fonts/*'
+                        '<%%= yeoman.dist %>/styles/fonts/{,*/}*.{eot,svg,ttf,woff}'
```
Comment (alrra) [inline app/templates/Gruntfile.js]: @felquis you should also add `.otf` ([OpenType](http://en.wikipedia.org/wiki/OpenType)) and `.ttc` ([TrueType Collection](http://en.wikipedia.org/wiki/TrueType))

Cc: @sindresorhus

Comment (sindresorhus) [inline app/templates/Gruntfile.js]: :+1:

Comment (felquis) [inline app/templates/Gruntfile.js]: I think this is unnecessary.

Font-squirrel provides to use:

```
TTF - Works in most browsers except IE and iPhone.
EOT - IE only. 
WOFF - Compressed, emerging standard. 
SVG - iPhone/iPad.
```

([ref](http://www.fontsquirrel.com/fonts/GoodDog)) See the tab "Webfont Kit"

Chris Coyier proposes to use TTF, EOT, WOFF and SVG ([ref](http://css-tricks.com/snippets/css/using-font-face/))

---

So, I think the better way to do this is using:

```
styles/fonts/{,*/}*.*
```

What you think about it?

Comment (alrra) [inline app/templates/Gruntfile.js]: > I think this is unnecessary.

@felquis the OpenType format is [quite used](http://www.paulirish.com/2009/bulletproof-font-face-implementation-syntax/).

Comment (sindresorhus) [inline app/templates/Gruntfile.js]: > styles/fonts/{,_/}_.*

Go for it. You shouldn't have anything other than fonts in the fonts folder anyways.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**